src/ray/core_worker/core_worker.cc

```diff
@@ -620,7 +616,12 @@ Status CoreWorker::CreateActor(const RayFunction &function,
   *return_actor_id = actor_id;
   TaskSpecification task_spec = builder.Build();
   PinObjectReferences(task_spec, TaskTransportType::RAYLET);
-  return raylet_client_->SubmitTask(task_spec);
+  // TODO(ekl) if we moved actor creation to use direct call tasks, then we won't
+  // need to manually resolve direct call args here.
+  resolver_->ResolveDependencies(task_spec, [this, task_spec]() {
```

Comment:
Just wondering, was there a test that was failing with the old version of this?

Reply:
Yeah, it causes rllib ES to hang, since we pass an object id to an actor constructor there.